### ISSUE TITLE
[SUGGESTION] CircularOutput _circular copy and clear

### ISSUE THREAD

**Issue by RolandasRazma** [raspberrypi/picamera2#519]
picamera v1 had ability to copy from  PiCameraCircularIO (copy_to) and clear it (clear) would be nice to have it in picamera v2 `CircularOutput` as well.

As a workaround I can access internal output._lock and output._circular but would be nice to have public API

**Comment by davidplowman** [comment]
@chrisruk Any thoughts on this one?

**Comment by chrisruk** [comment]
Yep, that sounds like it should be possible, to copy the contents of the circular buffer and clear it.  Will try
and look at this tomorrow.
